### PR TITLE
Add envs

### PR DESCRIPTION
**Pull request by ArtoLord**

### CODE REVIEW

File: lzy-api/src/main/proto/ai/lzy/v1/workflow/workflow.proto [hunk 1]
```diff
@@ -25,12 +25,14 @@ message Operation {
 
   string poolSpecName = 8;  // Label to execute operation in
 
+  /* optional */ map<string, string> envVariables = 9;  // Environment variables to run operation with
```
Comment (imakunin) [inline lzy-api/src/main/proto/ai/lzy/v1/workflow/workflow.proto:28]: let's rename to to `env`

File: model/src/main/proto/ai/lzy/v1/common/env.proto [hunk 2]
```diff
@@ -6,8 +6,16 @@ option java_outer_classname = "LME";
 option java_package = "ai.lzy.v1.common";
 
 message EnvSpec {
-  BaseEnv baseEnv = 1;
-  AuxEnv auxEnv = 2;
+  BaseEnv baseEnv = 1 [deprecated=true];  // TODO(artolord) remove after next minor
```
Comment (imakunin) [inline model/src/main/proto/ai/lzy/v1/common/env.proto:9]: remove it right now

File: model/src/main/proto/ai/lzy/v1/common/env.proto [hunk 3]
```diff
@@ -6,8 +6,16 @@ option java_outer_classname = "LME";
...
+  AuxEnv auxEnv = 2 [deprecated=true];
+
+  /* optional */ string dockerImage = 3;
+  /* optional */ map<string, string> envVariables = 4;
```
Comment (imakunin) [inline model/src/main/proto/ai/lzy/v1/common/env.proto:13]: just `env`

File: model/src/main/proto/ai/lzy/v1/common/env.proto [hunk 4]
```diff
@@ -6,8 +6,16 @@ option java_outer_classname = "LME";
...
+
+  oneof env {
```
Comment (imakunin) [inline model/src/main/proto/ai/lzy/v1/common/env.proto:15]: not `env` but `runtimeSpec`

File: model/src/main/proto/ai/lzy/v1/common/env.proto [hunk 5]
```diff
@@ -6,8 +6,16 @@ option java_outer_classname = "LME";
...
+    PythonEnv pyenv = 5;
```
Comment (imakunin) [inline model/src/main/proto/ai/lzy/v1/common/env.proto:16]: `PythonRuntimeSpec`

File: model/src/main/proto/ai/lzy/v1/common/env.proto [hunk 6]
```diff
@@ -6,8 +6,16 @@ option java_outer_classname = "LME";
...
+    ProcessEnv process_env = 6;
```
Comment (imakunin) [inline model/src/main/proto/ai/lzy/v1/common/env.proto:17]: `BashRuntimeSpec`

**Review by mrMakaronka — CHANGES_REQUESTED** [review]
No python unit tests on setting/merging env variables in workflow/op. Please add them to `LzyOpParamsTests`

**Comment by ArtoLord** [comment]
> No python unit tests on setting/merging env variables in workflow/op. Please add them to `LzyOpParamsTests`

Fixed

File: lzy-api/src/main/proto/ai/lzy/v1/workflow/workflow.proto [hunk 7]
```diff
@@ -25,12 +25,14 @@ message Operation {
...
+
   message SlotDescription {
     string path = 1;
     string storageUri = 2;
   }
 
-  message PythonEnvSpec {  // TODO(artolord) remove this and move to upper level
+  message PythonEnvSpec {
```
Comment (imakunin) [inline lzy-api/src/main/proto/ai/lzy/v1/workflow/workflow.proto:35]: maybe we should rename it to `PythonRunnerSpec` and rename `env` on line 22 to `runner`

File: model/src/main/proto/ai/lzy/v1/common/env.proto (hunk 4 above)
Comment (imakunin) [reply model/src/main/proto/ai/lzy/v1/common/env.proto:15]: or `runner`

**Review by imakunin — APPROVED** [review]

**Pull request merged**